components/TaskCard.vue

```diff
@@ -35,19 +51,82 @@ export default {
     task: null,
     loading: false,
   }),
+  computed: {
+    ...mapGetters({ requestInProcess: 'todos/getCompleteRequest' }),
+
+    formatDate() {
+      return this.$helper.formatDate(this.task.createdAt);
+    },
+    duration() {
+      return this.$helper.getDuration(
+        this.task.createdAt,
+        this.task.completedAt
+      );
+    },
+  },
+
   created() {
-    this.task = _.clone(this.cardData);
+    this.task = this.cardData;
   },
   methods: {
-    deleteTask() {
-      this.$store.dispatch('todos/deleteTask', this.task);
+    async markDone() {
+      this.loading = true;
+      await this.$store
+        .dispatch('todos/changeTaskState', this.task)
+        .then(() => {
+          if (!this.requestInProcess) {
+            this.loading = false;
+          }
+        });
+    },
+    async deleteTask() {
+      this.loading = true;
+      await this.$store.dispatch('todos/deleteTask', this.task).then(() => {
+        if (!this.requestInProcess) {
+          this.loading = false;
+        }
```

Comment:
Ditto. await automatically blocks the function execution until the promise is resolved. We can place the code of the `.then` block after the await.

Reply:
Resolved